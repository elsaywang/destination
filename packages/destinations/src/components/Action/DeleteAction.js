import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import Button from '@react/react-spectrum/Button';
import Delete from '@react/react-spectrum/Icon/Delete';
import Alert from '@react/react-spectrum/Icon/Alert';
import styles from './action.css';

class DeleteAction extends Component {
    renderTitle = () => (
        <Fragment>
            <Alert size="S" className={styles.deletionAlert} />
            <span className={styles.deletionText}>&nbsp; Delete Destination</span>
        </Fragment>
    );

    render() {
        const { destination, disabled, deleteDestination } = this.props;
        const { name } = destination;
        const boundDeleteClick = deleteDestination.bind(this, destination);

        return (
            !disabled && (
                <ModalTrigger>
                    <Button
                        data-test="action-delete-button"
                        label={''}
                        variant="action"
                        quiet
                        icon={<Delete size="S" />}
                        modaltrigger
                    />
                    <Dialog
                        data-test="action-delete-modal"
                        className={styles.deletionTextDialog}
                        modalcontent
                        title={this.renderTitle()}
                        confirmLabel="Delete"
                        size="S"
                        cancelLabel="Cancel"
                        onConfirm={boundDeleteClick}
                        variant="destructive">
                        <span>
                            Are you sure you want to delete this <strong>{name}</strong>{' '}
                            destination?
                            <br />
                        </span>
                    </Dialog>
                </ModalTrigger>
            )
        );
    }
}

DeleteAction.defaultProps = {
    disabled: false,
    deleteDestination: () => {
        //no-ops
    },
};

DeleteAction.protoTypes = {
    deleteDestination: PropTypes.func, //TODO
    destination: PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
    }),
};

export default DeleteAction;
