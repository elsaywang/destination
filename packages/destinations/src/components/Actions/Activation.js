import React from 'react';
import PropTypes from 'prop-types';
import styles from './action.css';
import AddAccountModal from '../AddAccountModal';

const Activation = ({ authentication, disabled, addAccount }) => (
    <AddAccountModal
        message={authentication.expireIn === 'Expired' ? 'Reactivate' : 'Renew'}
        quiet
        className={styles.activationButton}
        disabled={disabled}
        addAccount={addAccount}
        authentication={authentication}
    />
);

Activation.defaultProps = {
    disabled: false,
    addAccount: () => {
        //no-ops
    },
};

Activation.propTypes = {
    //TODO: AddAccountModal may need to add corresponding props value
    authentication: PropTypes.shape({
        adAccountId: PropTypes.string,
        accountName: PropTypes.string,
        expireIn: PropTypes.string,
    }),
    addAccount: PropTypes.func, //TODO
    disabled: PropTypes.bool,
};

export default Activation;
