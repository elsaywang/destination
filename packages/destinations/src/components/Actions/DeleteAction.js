import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import ModalTrigger from '@react/react-spectrum/ModalTrigger';
import Dialog from '@react/react-spectrum/Dialog';
import Button from '@react/react-spectrum/Button';
import Delete from '@react/react-spectrum/Icon/Delete';
import Alert from '@react/react-spectrum/Icon/Alert';
import requiredIf from 'react-required-if';
import styles from './action.css';

class DeleteAction extends Component {
    renderTitle = () => (
        <Fragment>
            <Alert size="S" className={styles.deletionAlert} />
            <span className={styles.deletionText}>
                &nbsp; Delete {this.props.isForDestination ? `Destination` : `Authentication`}
            </span>
        </Fragment>
    );

    render() {
        const { disabled, handleDeleteAction, isForDestination } = this.props;
        const { name, destinationId } = isForDestination && this.props.destination;
        const { accountName, adAccountId } = !isForDestination && this.props.authentication;

        return (
            !disabled && (
                <ModalTrigger>
                    <Button
                        data-test="action-delete-button"
                        label={isForDestination ? '' : 'Delete'}
                        variant="action"
                        quiet
                        icon={isForDestination && <Delete size="S" />}
                        modaltrigger
                    />
                    <Dialog
                        data-test="action-delete-modal"
                        className={styles.deletionTextDialog}
                        modalcontent
                        title={this.renderTitle()}
                        confirmLabel="Delete"
                        size="S"
                        cancelLabel="Cancel"
                        onConfirm={() =>
                            handleDeleteAction(isForDestination ? destinationId : adAccountId)
                        }
                        variant="destructive">
                        <span>
                            Are you sure you want to delete this{' '}
                            <strong>{isForDestination ? name : accountName}</strong>{' '}
                            {isForDestination ? `destination` : `authentication`} ?
                            <br />
                        </span>
                    </Dialog>
                </ModalTrigger>
            )
        );
    }
}

DeleteAction.defaultProps = {
    disabled: false,
    handleDeleteAction: () => {
        //no-ops
    },
};

DeleteAction.propTypes = {
    isForDestination: PropTypes.bool,
    handleDeleteAction: PropTypes.func,
    destination: requiredIf(
        PropTypes.shape({
            destinationId: PropTypes.number,
            name: PropTypes.string,
        }),
        ({ isForDestination }) => isForDestination,
    ),
    authentication: requiredIf(
        PropTypes.shape({
            adAccountId: PropTypes.string,
            accountName: PropTypes.string,
        }),
        ({ isForDestination }) => !isForDestination,
    ),
};

export default DeleteAction;
