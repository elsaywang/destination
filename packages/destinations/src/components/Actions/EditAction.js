import React from 'react';
import PropTypes from 'prop-types';
import Button from '@react/react-spectrum/Button';
import Edit from '@react/react-spectrum/Icon/Edit';
import { editDestinationUrl } from '../../utils/urls';

const EditAction = ({ destination, disabled, editDestination }) => (
    <Button
        data-test="action-edit-button"
        label={''}
        variant="action"
        quiet
        onClick={editDestination}
        icon={<Edit size="S" />}
        disabled={disabled}
        target="_blank"
        element="a"
        rel="noopener"
        href={editDestinationUrl(destination.destinationId)}
    />
);

EditAction.defaultProps = {
    disabled: false,
    editDestination: () => {
        //no-ops
    },
};

EditAction.propTypes = {
    destination: PropTypes.shape({
        destinationId: PropTypes.number,
        name: PropTypes.string,
    }),
    editDestination: PropTypes.func, //TODO
};

export default EditAction;
