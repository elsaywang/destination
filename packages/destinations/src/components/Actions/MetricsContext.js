import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Heading from '@react/react-spectrum/Heading';
import styles from './action.css';
import { metricsHeadings } from '../../constants/metrics';

const MetricsContext = ({ type, ...metricsValues }) => {
    return metricsHeadings
        .filter(({ integratedType }) => integratedType === type)
        .map(({ heading, key }) => (
            <Fragment key={key}>
                <Heading
                    variant="subtitle2"
                    className={classNames(styles.contextHeading, {
                        [styles.lifeTimeAddressableAudience]:
                            `${key}` === `lifetimeAddressableAudience`,
                    })}>
                    {heading}
                </Heading>
                <span className={styles.contextData}>{metricsValues[key].toLocaleString()}</span>
            </Fragment>
        ));
};

MetricsContext.defaultProps = {};

MetricsContext.propTypes = {
    type: PropTypes.oneOf(['People-Based', 'Device-Based']),
    shareableAudience: PropTypes.number,
    addressableAudience: PropTypes.number,
    matchRate: PropTypes.string,
    lifetimeAddressableAudience: PropTypes.number,
};

export default MetricsContext;
