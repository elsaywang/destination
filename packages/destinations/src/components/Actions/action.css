:local(.deletionAlert){
  color:var(--spectrum-semantic-notice-color-state-focus);
  margin: -0.5rem 0 -0.15rem 0.1rem
}
:local(.deletionText){
  font-weight: 300;
  font-style:normal;
  font-size:1.25rem;
}
:local(.deletionTextDialog){
  padding: 0.985rem
}
:local(.metricsTitle){
  font-weight: var(--spectrum-global-font-weight-black);
  font-style:normal;
  font-size:var(--spectrum-body-small-font-size-default);
  color:var(--spectrum-global-color-grey500);
}
:local(.contextHeading){
  font-weight: var(--spectrum-global-font-weight-light);
  margin-top: -0.25rem;
  font-style:normal;
  font-size:var(--spectrum-global-font-size-medium);
}
:local(.contextData){
  display: inline-block;
  color:var(--spectrum-semantic-positive-color-status);
  font-size:var(--spectrum-global-font-size-xlarge);
  font-weight: var(--spectrum-global-font-weight-light);
  margin-top: -0.5rem;
}
:local(.lifeTimeAddressableAudience){
  margin-top: 0.85rem;
}
:local(.activationButton){
  font-weight: var(--spectrum-global-font-weight-normal);
  max-width: 1.25rem;
}

:local(.actions){
  margin-left: -0.95rem;
}
