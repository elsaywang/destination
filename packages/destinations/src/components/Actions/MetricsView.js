import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import requiredIf from 'react-required-if';
import Button from '@react/react-spectrum/Button';
import Popover from '@react/react-spectrum/Popover';
import OverlayTrigger from '@react/react-spectrum/OverlayTrigger';
import GraphBarVertical from '@react/react-spectrum/Icon/GraphBarVertical';
import MetricsContext from './MetricsContext';
import styles from './action.css';

const MetricsView = ({ destination }) => {
    const { addressableAudienceMetrics, destinationType } = destination;

    const metricsProps = _.merge(
        {},
        destinationType === 'S2S' && {
            type: 'Device-Based',
            addressableAudience: addressableAudienceMetrics.clientAddressableAudience30Day,
            matchRate: '0%', //TODO: need to get the correct formula to calculate
            lifetimeAddressableAudience:
                addressableAudienceMetrics.platformAddressableAudienceLifetime,
        },
        destinationType === 'PEOPLE_BASED' && {
            type: 'People-Based',
            shareableAudience: 34785, // TODO: fetch from API for PEOPLE_BASED
        },
    );

    const { type } = metricsProps;

    const renderTitle = () => (
        <span className={styles.metricsTitle}>{`${type.toUpperCase()} PLATFORM`}</span>
    );

    return (
        <OverlayTrigger trigger="click" placement="left">
            <Button label="" variant="action" quiet icon={<GraphBarVertical size="S" />} />
            <Popover title={renderTitle()}>
                <MetricsContext {...metricsProps} />
            </Popover>
        </OverlayTrigger>
    );
};

MetricsView.defaultProps = {
    disabled: false,
};

MetricsView.propTypes = {
    disabled: PropTypes.bool,
    destination: PropTypes.shape({
        destinationId: PropTypes.number,
        name: PropTypes.string,
        destinationType: PropTypes.oneOf(['PEOPLE_BASED', 'S2S']).isRequired,
        addressableAudienceMetrics: requiredIf(
            PropTypes.shape({
                platformAddressableAudienceLifetime: PropTypes.number,
                clientAddressableAudience30Day: PropTypes.number,
            }),
            ({ destinationType }) => destinationType === 'S2S',
        ),
        shareableAudience: requiredIf(
            PropTypes.number,
            ({ destinationType }) => destinationType === 'PEOPLE_BASED',
        ),
    }),
};

export default MetricsView;
