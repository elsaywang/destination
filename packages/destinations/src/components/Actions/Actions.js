import React, { Fragment } from 'react';
import requiredIf from 'react-required-if';
import PropTypes from 'prop-types';
import DeleteAction from './DeleteAction';
import Activation from './Activation';
import EditAction from './EditAction';
import MetricsView from './MetricsView';
import styles from './action.css';

const Actions = ({ isForDestination, disabled, ...actionsProps }) => {
    const { destination, showMetrics } = isForDestination && actionsProps;
    const { authentication } = !isForDestination && actionsProps;

    const renderActionsForDestination = isForDestination && (
        <Fragment>
            <EditAction destination={destination} disabled={disabled} />
            <DeleteAction destination={destination} isForDestination={isForDestination} />
            {showMetrics && <MetricsView destination={destination} disabled={disabled} />}
        </Fragment>
    );

    const renderActionsForAuthentication = !isForDestination && (
        <Fragment>
            <Activation authentication={authentication} disabled={disabled} />
            <DeleteAction authentication={authentication} isForDestination={isForDestination} />
        </Fragment>
    );

    return (
        <div className={styles.actions}>
            {renderActionsForDestination}
            {renderActionsForAuthentication}
        </div>
    );
};

Actions.defaultProps = {
    disabled: false,
};

Actions.propTypes = {
    destination: requiredIf(
        PropTypes.shape({
            destinationId: PropTypes.number,
            name: PropTypes.string,
            destinationType: PropTypes.oneOf(['PEOPLE_BASED', 'S2S', 'PUSH', 'ADS', 'ANALYTICS']),
        }),
        ({ isForDestination }) => isForDestination,
    ),
    authentication: requiredIf(
        PropTypes.shape({
            adAccountId: PropTypes.string,
            accountName: PropTypes.string,
        }),
        ({ isForDestination }) => !isForDestination,
    ),
    isForDestination: PropTypes.bool,
    disabled: PropTypes.bool,
    showMetrics: requiredIf(PropTypes.bool, ({ isForDestination }) => isForDestination),
    handleDeleteAction: PropTypes.func,
};
export default Actions;
