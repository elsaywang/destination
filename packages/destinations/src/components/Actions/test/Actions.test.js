import React from 'react';
import { shallow } from 'enzyme';
import Actions from '../Actions';
import DeleteAction from '../DeleteAction';
import EditAction from '../EditAction';
import MetricsView from '../MetricsView';

describe('<Actions/> component', () => {
    describe('rendering', () => {
        const mockFn = jest.fn();
        const props = {
            disabled: false,
            destination: {
                id: 20008,
                name: 'card',
                type: 'People-Based',
            },
            showMetrics: false,
        };
        const wrapper = shallow(<Actions {...props} />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders only <DeleteAction/> and <EditAction/> if `showMetrics` prop is false', () => {
            expect(wrapper.find(DeleteAction).exists()).toBeTruthy();
            expect(wrapper.find(EditAction).exists()).toBeTruthy();
            expect(wrapper.find(MetricsView).exists()).toBeFalsy();
        });

        it('renders  <DeleteAction/>, <EditAction/> and <MetricsView/> if `showMetrics` prop is true', () => {
            wrapper.setProps({ showMetrics: true });
            expect(wrapper.find(DeleteAction).exists()).toBeTruthy();
            expect(wrapper.find(EditAction).exists()).toBeTruthy();
            expect(wrapper.find(MetricsView).exists()).toBeTruthy();
        });
    });
});
