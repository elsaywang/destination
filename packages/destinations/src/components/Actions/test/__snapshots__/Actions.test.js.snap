// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions/> component rendering when \`isForDestination\`  matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Actions
    destination={
        Object {
            "destinationType": "PEOPLE_BASED",
            "id": 20008,
            "name": "card",
            "shareableAudience": 909,
          }
    }
    disabled={false}
    isForDestination={true}
    showMetrics={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown>
          <EditAction
                    destination={
                              Object {
                                        "destinationType": "PEOPLE_BASED",
                                        "id": 20008,
                                        "name": "card",
                                        "shareableAudience": 909,
                                      }
                    }
                    disabled={false}
                    editDestination={[Function]}
          />
          <DeleteAction
                    destination={
                              Object {
                                        "destinationType": "PEOPLE_BASED",
                                        "id": 20008,
                                        "name": "card",
                                        "shareableAudience": 909,
                                      }
                    }
                    disabled={false}
                    handleDeleteAction={[Function]}
                    isForDestination={true}
          />
</Unknown>,
        false,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <EditAction
              destination={
                            Object {
                                          "destinationType": "PEOPLE_BASED",
                                          "id": 20008,
                                          "name": "card",
                                          "shareableAudience": 909,
                                        }
              }
              disabled={false}
              editDestination={[Function]}
/>,
            <DeleteAction
              destination={
                            Object {
                                          "destinationType": "PEOPLE_BASED",
                                          "id": 20008,
                                          "name": "card",
                                          "shareableAudience": 909,
                                        }
              }
              disabled={false}
              handleDeleteAction={[Function]}
              isForDestination={true}
/>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "destination": Object {
                "destinationType": "PEOPLE_BASED",
                "id": 20008,
                "name": "card",
                "shareableAudience": 909,
              },
              "disabled": false,
              "editDestination": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "destination": Object {
                "destinationType": "PEOPLE_BASED",
                "id": 20008,
                "name": "card",
                "shareableAudience": 909,
              },
              "disabled": false,
              "handleDeleteAction": [Function],
              "isForDestination": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": Symbol(react.fragment),
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown>
            <EditAction
                        destination={
                                    Object {
                                                "destinationType": "PEOPLE_BASED",
                                                "id": 20008,
                                                "name": "card",
                                                "shareableAudience": 909,
                                              }
                        }
                        disabled={false}
                        editDestination={[Function]}
            />
            <DeleteAction
                        destination={
                                    Object {
                                                "destinationType": "PEOPLE_BASED",
                                                "id": 20008,
                                                "name": "card",
                                                "shareableAudience": 909,
                                              }
                        }
                        disabled={false}
                        handleDeleteAction={[Function]}
                        isForDestination={true}
            />
</Unknown>,
          false,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <EditAction
                destination={
                                Object {
                                                "destinationType": "PEOPLE_BASED",
                                                "id": 20008,
                                                "name": "card",
                                                "shareableAudience": 909,
                                              }
                }
                disabled={false}
                editDestination={[Function]}
/>,
              <DeleteAction
                destination={
                                Object {
                                                "destinationType": "PEOPLE_BASED",
                                                "id": 20008,
                                                "name": "card",
                                                "shareableAudience": 909,
                                              }
                }
                disabled={false}
                handleDeleteAction={[Function]}
                isForDestination={true}
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "destination": Object {
                  "destinationType": "PEOPLE_BASED",
                  "id": 20008,
                  "name": "card",
                  "shareableAudience": 909,
                },
                "disabled": false,
                "editDestination": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "destination": Object {
                  "destinationType": "PEOPLE_BASED",
                  "id": 20008,
                  "name": "card",
                  "shareableAudience": 909,
                },
                "disabled": false,
                "handleDeleteAction": [Function],
                "isForDestination": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": Symbol(react.fragment),
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Actions/> component rendering when \`isForDestination\` is false matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Actions
    authentication={
        Object {
            "accountName": "test",
            "adAccountId": "1",
            "expireIn": "Expired",
          }
    }
    disabled={false}
    isForDestination={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <Unknown>
          <Activation
                    addAccount={[Function]}
                    authentication={
                              Object {
                                        "accountName": "test",
                                        "adAccountId": "1",
                                        "expireIn": "Expired",
                                      }
                    }
                    disabled={false}
          />
          <DeleteAction
                    authentication={
                              Object {
                                        "accountName": "test",
                                        "adAccountId": "1",
                                        "expireIn": "Expired",
                                      }
                    }
                    disabled={false}
                    handleDeleteAction={[Function]}
                    isForDestination={false}
          />
</Unknown>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Activation
              addAccount={[Function]}
              authentication={
                            Object {
                                          "accountName": "test",
                                          "adAccountId": "1",
                                          "expireIn": "Expired",
                                        }
              }
              disabled={false}
/>,
            <DeleteAction
              authentication={
                            Object {
                                          "accountName": "test",
                                          "adAccountId": "1",
                                          "expireIn": "Expired",
                                        }
              }
              disabled={false}
              handleDeleteAction={[Function]}
              isForDestination={false}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addAccount": [Function],
              "authentication": Object {
                "accountName": "test",
                "adAccountId": "1",
                "expireIn": "Expired",
              },
              "disabled": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "authentication": Object {
                "accountName": "test",
                "adAccountId": "1",
                "expireIn": "Expired",
              },
              "disabled": false,
              "handleDeleteAction": [Function],
              "isForDestination": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <Unknown>
            <Activation
                        addAccount={[Function]}
                        authentication={
                                    Object {
                                                "accountName": "test",
                                                "adAccountId": "1",
                                                "expireIn": "Expired",
                                              }
                        }
                        disabled={false}
            />
            <DeleteAction
                        authentication={
                                    Object {
                                                "accountName": "test",
                                                "adAccountId": "1",
                                                "expireIn": "Expired",
                                              }
                        }
                        disabled={false}
                        handleDeleteAction={[Function]}
                        isForDestination={false}
            />
</Unknown>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Activation
                addAccount={[Function]}
                authentication={
                                Object {
                                                "accountName": "test",
                                                "adAccountId": "1",
                                                "expireIn": "Expired",
                                              }
                }
                disabled={false}
/>,
              <DeleteAction
                authentication={
                                Object {
                                                "accountName": "test",
                                                "adAccountId": "1",
                                                "expireIn": "Expired",
                                              }
                }
                disabled={false}
                handleDeleteAction={[Function]}
                isForDestination={false}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "addAccount": [Function],
                "authentication": Object {
                  "accountName": "test",
                  "adAccountId": "1",
                  "expireIn": "Expired",
                },
                "disabled": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "authentication": Object {
                  "accountName": "test",
                  "adAccountId": "1",
                  "expireIn": "Expired",
                },
                "disabled": false,
                "handleDeleteAction": [Function],
                "isForDestination": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
